---
title: C28125 |Microsoft Docs
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C28125
helpviewer_keywords:
- C28125
ms.assetid: 3f7b8db4-6a64-4480-919e-3f3ceca7f948
caps.latest.revision: 6
author: mikeblome
ms.author: mblome
manager: ghogen
ms.openlocfilehash: 1f7733ed4fcac9b6d1936b3f8e52a1c03d78a6cb
ms.sourcegitcommit: af428c7ccd007e668ec0dd8697c88fc5d8bca1e2
ms.translationtype: MT
ms.contentlocale: zh-CN
ms.lasthandoff: 11/16/2018
ms.locfileid: "51788415"
---
# <a name="c28125"></a>C28125
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

警告 C28125： 必须在一个 try / except 块中调用函数  
  
 该驱动程序正在调用的函数必须从调用在一个 try / except 块中，如[ProbeForRead](http://msdn.microsoft.com/library/windows/hardware/ff559876.aspx)， [ProbeForWrite](http://msdn.microsoft.com/library/windows/hardware/ff559879.aspx)， [MmProbeAndLockPages](http://msdn.microsoft.com/library/windows/hardware/ff554664.aspx)。  
  
## <a name="example"></a>示例  
 下面的代码示例将生成此警告：  
  
```  
ProbeForRead(addr, len, 4);  
```  
  
 下面的代码示例可避免此警告：  
  
```  
  
__try  
{  
    ProbeForRead(addr, len, 4);  
}  
__except(EXCEPTION_EXECUTE_HANDLER)  
{  
    Status = GetExceptionCode();  
    ... report error status  
}  
```



